contracts/polyx-vesting/lib.rs

```diff
@@ -0,0 +1,253 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+mod polyx_vesting {
+    use ink_storage::traits::SpreadAllocate;
+
+    /// Defines the storage of your contract.
+    #[ink(storage)]
+    #[derive(SpreadAllocate)]
+    pub struct PolyxVesting {
+        released: Balance,
+        beneficiary: AccountId,
+        start: Timestamp,
+        duration: Timestamp,
+    }
+
+    /// Event emitted when Polyx is released.
+    #[ink(event)]
+    pub struct PolyxReleased {
+        value: Balance,
+    }
+
+    /// The error types.
+    #[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Error {
+        /// Returned if not enough balance to fulfill a request is available.
+        InsufficientBalance,
+        /// Funds are not released as yet.
+        FundsNotReleased,
+        /// Invalid start timestamp
+        InvalidStartTimestamp,
+        /// Duration Overflow
+        DurationOverflow,
+        /// Invalid timestamp
+        InvalidTimestamp,
+        /// Timestamp Overflow
+        TimestampOverflow,
+    }
+
+    /// The contract result type.
+    pub type Result<T> = core::result::Result<T, Error>;
+
+    impl PolyxVesting {
+        /// Constructor
+        #[ink(constructor, payable)]
+        pub fn new(
+            beneficiary_address: AccountId,
+            start_timestamp: Timestamp,
+            duration_milli_seconds: Timestamp,
+        ) -> Result<Self> {
+            ink_lang::utils::initialize_contract(|contract| {
```

Comment:
Ink! <=3.4 only supports `Self` as a return type from constructors.  But they have fixed this in version 4.0: https://github.com/paritytech/ink/pull/1446

Polymesh doesn't support Ink! 4.0 until we update the Substrate monthly PR.  So for now just add a comment with a note about this with the link to that Ink! PR and use `unwrap` to return `Self` here.